diff -Naur old/Makefile new/Makefile
--- old/Makefile	2021-01-03 07:31:56.000000000 +0800
+++ new/Makefile	2021-03-25 13:49:20.897014538 +0800
@@ -183,7 +183,7 @@
          CPUFLAGS += -mcpu=cortex-a15 -mtune=cortex-a15.cortex-a7 -mfpu=neon-vfpv4
       else
          # ODROID-U3, U2, X2 & X
-          CPUFLAGS += -mcpu=cortex-a9 -mfpu=neon
+          CPUFLAGS += -mcpu=cortex-a7 -mfpu=neon
       endif
    endif
    
